embassy-stm32/src/dma/mod.rs

```diff
@@ -59,7 +59,7 @@ pub(crate) mod sealed {
         unsafe fn start_write_repeated<W: super::Word>(
             &mut self,
             request: Request,
-            repeated: W,
+            repeated: &[W; 1],
```

Comment:
We should use raw pointers for consistency with the other methods: `repeated: *const W`.

(The reason to use raw pointers is that the buffer is required to live for the whole transfer, using references gives the impression that living for the duration of the function call only is enough)